system/CLI/CLI.php

```diff
@@ -264,7 +272,7 @@ public static function prompt(string $field, $options = null, string $validation
 		// Read the input from keyboard.
 		$input = trim(static::input()) ?: $default;
 
-		if (isset($validation))
+		if ($validation)
 		{
 			while (! static::validate($field, $input, $validation))
```

Comment:
$validation->setRules() which is used by validate expects string rules.

```suggestion
			$validation = trim(implode('|', $validation), '|');

			while (! static::validate($field, $input, $validation))
```

Reply:
It must have been a mistake on my part as I tested this with several rules.
I would not drop support of array-based validation rules since it is required for features like anonymous functions as rules. Instead i'd chanche `validate` to internaly use `setRules` which supports array-based rules?

Reply:
That would not be possible at the moment without a breaking change. Validation::setRule has this signature
```
public function setRule(string $field, string $label = null, string $rules, array $errors = [])
```
Removing the string typehint would be breaker for existing user implementations overriding the method since this is declared public.

Reply:
Sorry, i wasnt clear about that. I'd wanted to change to use setRule**s** which can handle that.
```php
$validation->setRules([$field => [
    'label' => $label,
    'rules' => $rules,
]]);
```

Reply:
Ah. So you'd be swapping `setRule` with `setRules`. Fair enough. That's ok with me.